src/main/clojure/clara/macros.clj

```diff
@@ -180,7 +180,8 @@
               (com/compile-action all-bindings
                                   ;; Using private function for now as a workaround.
                                   (if (:ns-name production)
-                                    (binding [*ns* (the-ns (:ns-name production))]
+                                    (binding [*ns* (the-ns (:ns-name production))
+                                              cljs.analyzer/*cljs-ns* (:ns-name production)]
```

Comment:
[Clojure's the-ns](http://clojuredocs.org/clojure.core/the-ns) throws an exception if the namespace isn't found.  I was very surprised that this didn't fail, since I expected the test rules ClojureScript namespace to not exist in the Clojure environment of the ClojureScript compiler.  I put in some debug statements and it appears that the ClojureScript compiler creates an empty Clojure namespace of the same name, at least in that case.  I dug around a bit and found some calls like this in Clojure in the [ClojureScript analyzer](https://github.com/clojure/clojurescript/blob/master/src/main/clojure/cljs/analyzer.cljc):

````
(create-ns *cljs-ns*)
````

I haven't dug around in the ClojureScript compiler to understand why it would do this, but it seems like an implementation detail that could break.  Since direct use of this namespace is deprecated in favor of use through rules.cljc, we probably won't have usage of it in Clojure; see [defsession in clara.rules/defsession](https://github.com/cerner/clara-rules/blob/0.16.1/src/main/clojure/clara/rules.cljc#L357) and its switching on whether Clojure or ClojureScript is being compiled.  Nevertheless it is easy enough to support both here in a safe way that my inclination is to keep doing so for the moment pending a more complete discussion of fully "unsupporting" this namespace in Clojure.

That is:

````
(if (:ns-name production)
     (if (compiling-cljs?)
         (binding [cljs.analyzer/*cljs-ns* (:ns-name-production)]
                 (resolve-vars)
         (binding [*ns* (the-ns (:ns-name production))]
                 (resolve-vars))
   .....
````